plugins/woocommerce/src/Blocks/AssetsController.php

```diff
@@ -44,33 +45,40 @@ protected function init() { // phpcs:ignore WooCommerce.Functions.InternalInject
 		add_action( 'wp_enqueue_scripts', array( $this, 'update_block_settings_dependencies' ), 100 );
 		add_action( 'admin_enqueue_scripts', array( $this, 'update_block_settings_dependencies' ), 100 );
 		add_filter( 'js_do_concat', array( $this, 'skip_boost_minification_for_cart_checkout' ), 10, 2 );
+
+		if ( Features::is_enabled( 'experimental-iapi-runtime' ) ) {
+			// Run after the WordPress iAPI runtime has been registered by setting a lower priority.
+			add_filter( 'wp_default_scripts', array( $this, 'deregister_core_iapi_runtime' ), 20 );
+		}
 	}
 
 	/**
-	 * Register script modules.
+	 * De-register the iAPI runtime registered by WordPress core/Gutenberg, wllowing us to register our own version of the iAPI runtime.
 	 */
-	public function register_script_modules() {
-		// Right now we only have one script modules build for supported interactivity API powered block front-ends.
-		// We generate a combined asset file for that via DependencyExtractionWebpackPlugin to make registration more
-		// efficient.
-		$asset_data = $this->api->get_asset_data(
-			$this->api->get_block_asset_build_path( 'interactivity-blocks-frontend-assets', 'php' )
-		);
+	public function deregister_core_iapi_runtime() {
```

Comment:
Perhaps this function should be called reregister, since it is not only unregistering the interactivity API from Core, but also registering the WooCommerce interactivity API?

```suggestion
	public function reregister_core_iapi_runtime() {
```

Reply:
good suggestion, thanks https://github.com/woocommerce/woocommerce/pull/57611/commits/951e5fc6b875086f10cd3530fd3ac6ed2aef7ce2